libr/arch/p/arm/plugin_cs.c

```diff
@@ -2495,7 +2495,7 @@ static int analop_esil(RArchSession *as, RAnalOp *op, ut64 addr, const ut8 *buf,
 		break;
 	case ARM_INS_BL:
 	case ARM_INS_BLX:
-		r_strbuf_append (&op->esil, "pc,lr,:=,");
+		r_strbuf_appendf (&op->esil, "pc,%d,+,lr,:=,", thumb);
```

Comment:
```suggestion
		r_strbuf_appendf (&op->esil, "pc,lr,:=,%d,tf,:=", thumb);
```

Reply:
what do you think about this solution? its working?

Reply:
if not we should probably considering checking for the thumb flag in the register set which is more correct for esil cnoncepts than using the hacky lower bit thing in the program counter

Reply:
Sorry, I got sidetracked from all this and am abroad at the moment so can't do much, but i'll defer to your experience here